backend/mlarchive/tests/archive/search_api.py

```diff
@@ -16,9 +16,106 @@ def test_ensure_index(settings):
     assert settings.ELASTICSEARCH_INDEX_NAME == 'test-mail-archive'
 
 
+# ------------------------------
+# Core Search
+# ------------------------------
+
+
+@pytest.mark.django_db(transaction=True)
+def test_form_text_frm_name(rf, client, search_api_messages):
+    '''From field term, name'''
+    request = rf.get('/arch/search/?q=zaphod')
+    request.user = AnonymousUser()
+    data = QueryDict('q=zaphod')
+    form = AdvancedSearchForm(data=data, request=request)
+    search = search_from_form(form)
+    results = search.execute()
+    print(search.to_dict())
+    assert len(results) == 1
+    ids = [h.msgid for h in results]
+    assert sorted(ids) == ['api001']
+
+
+@pytest.mark.django_db(transaction=True)
+def test_form_text_frm_domain(rf, client, search_api_messages):
+    '''From field term, email domain'''
+    request = rf.get('/arch/search/?q=example.com')
+    request.user = AnonymousUser()
+    data = QueryDict('q=example.com')
+    form = AdvancedSearchForm(data=data, request=request)
+    search = search_from_form(form)
+    results = search.execute()
+    print(search.to_dict())
+    assert len(results) == 4
+    ids = [h.msgid for h in results]
+    assert sorted(ids) == ['api001', 'api002', 'api003', 'api004']
+
+
+@pytest.mark.skip(reason='Does not break out domain parts, which seems ok')
```

Comment:
It's not clear why this is here as a skip. 

If it's to document that a search for example returns nothing, why is the test asserting there should be 1?

Further, if that's the point (that the test here would _pass_ if you changed the assert to expect 0 messages),
I think this will surprise people - will search right now not find cisco if the body of the message contains cisco.com (but not cisco by itself)?

Or have I missed the point?

Reply:
I left this skipped test in as something of a placeholder. I recognize that it could be surprising that searches for “cisco” do not match joe@cisco.com. I was reminded of this behavior after writing these tests, but feel it might be more appropriate to address as a separate issue. The cause of the behaviour is that we use the Standard Tokenizer, https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-tokenizers.html, which will not break words on the period (FULL STOP) per WB6/WB7 here: https://unicode.org/reports/tr29/#WB5. This is already the case for searches of the message body and specified “from:” header. It is not clear that one of the other tokenizers is a better fit. One possible solution is to use a character filter “s/\./ /g“, which will have some side effects. Need to consider this some more.